src/main/java/com/WebController/EventController.java

```diff
@@ -1,66 +1,65 @@
 package com.WebController;
 
-import com.Model.Patient;
-import com.Model.PatientEventAssociation;
-import com.Repository.AssociationRepository;
-import com.Repository.EventRepository;
-import com.Repository.PatientRepository;
-import com.Model.Event;
+import com.Model.*;
+import com.Repository.*;
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.stereotype.Controller;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
 import org.springframework.web.bind.annotation.*;
-
+import java.security.Principal;
 import java.util.ArrayList;
 import java.util.List;
 
-@Controller
+@RestController
 public class EventController {
 
-    @Autowired
-    private PatientRepository patientRepo;
-
     @Autowired
     private EventRepository eventRepo;
 
     @Autowired
-    private AssociationRepository assoicationRepo;
-
-    @GetMapping("/")
-    public String homePage() {
-        return "index";
-    }
+    private UserRepository userRepo;
 
+    /**
+     * Handler for patients creation.
+     * User create a patient in front end and associate patient with multiple events.
+     * Data posted with JSON string and parsed to Event objects directly.
+     * Event duplication should be avoided.
+     */
     @PostMapping(value = "/patientinfo", consumes = "application/json")
-    @ResponseBody
-    public String receivePatient (@RequestBody List<Event> events){
-        Patient tempPatient = new Patient();
-        List<Event> eventList = new ArrayList<>();
-        Iterable<Event> allEvents = this.eventRepo.findAll();
-        allEvents.forEach(eventList::add);
+    public List<Event> receivePatient (@RequestBody List<Event> events, Principal principal){
```

Comment:
this is much better! I think in our next iteration we should allow the adding of one event at a time. The way we could do it is that we first create a patient, we return a session id that we associate to that particular patient, and for every event during the session we retrieve the session id and use it to determine who this event should be associated with.